.github/workflows/ci-cd.yml

```diff
@@ -147,6 +147,7 @@ jobs:
         with:
           python-version: 3.11.2
       - run: pip install poetry
+      - run: poetry lock --no-update
```

Comment:
This is for syncing the lock file in the run tests, this will only sync the lock file if a new dependency is added in the pyproject.toml, for PRs adding/modifying the dependencies , i suppose its important.